### ISSUE TITLE
Exclude particular selections from calculation

### ISSUE THREAD

**Issue by MarHelmer** [phg49389/drg-weapons-calculator#14]
**Is your feature request related to a problem? Please describe.**
It would be awesome if we could exclude mods or OCs from the analysis. For instance, if I want to figure out the most targets the PGL can hit without Fat Boy, I don't have a way to do that other than manually re-running the analysis with each non-Fat Boy OC pre-selected.

**Describe the solution you'd like**
When using the "Best Practices (Subset)" option, the calculation processes for the displayed weapon using the currently selected mods and OCs.  This exclusionary filter, when added to particular mod(s) or OC(s), would prevent them from membership in the solution set when using a Subset calculation on the displayed weapon.

**Comment by phg49389** [comment]
I did some tinkering with the GUI buttons, and I was able to prototype having the Overclock buttons have a translucent red placed over them for when they get ignored, and I got most of the MouseListener implemented. I ran into an issue when trying to listen for right-clicks, as it made the left-click behavior really wonky. My original idea to right-click a button to ignore it shouldn't be pursued further, imo. Now I need to figure out a new way to toggle whether an OC (and eventually mod) should be ignored. Saving a code snippet so I don't have to re-figure it out later:

```
                // Paint this with a translucent red when it's not eligible for Best Combinations (Subset)
		if (ignored) {
			Color translucentRed = new Color(156.0f/255.0f, 20.0f/255.0f, 20.0f/255.0f, 0.5f);
			g2.setPaint(translucentRed);
			g2.fill(border);
		}
```

**Comment by MarHelmer** [comment]
An idea: cycle through options when left-clicked between unselected, selected, and exclude-selected.

**Comment by phg49389** [comment]
According to `git diff --stat`: 24 files changed, 848 insertions(+), 940 deletions(-)

This is probably the biggest change I've had to do for a single feature -- 1/20th of the entire codebase. The original issue I ran into with listening for right clicks was that I was using the recommended "only count single clicks" from SO. Once I removed that check, then both left-clicks and right-clicks felt crisp again. Additionally, by changing the listener from an ActionPerformed on buttonclick to MouseEvent on mouseclicks, I was able to change the buttons to only do stuff when the click is inside the visual borders. Still pops up HoverText in between buttons, sadly.

Feature added, will be in next release. Closing issue.
